packages/lit-dev-content/src/components/litdev-tutorial.ts

```diff
@@ -69,89 +93,217 @@ export class LitDevTutorial extends LitElement {
   @state()
   private _preloadedHtml?: {idx: number; promise: Promise<string>};
 
+  /**
+   * Tutorial manifest json file
+   */
+  @state()
+  private _manifest: TutorialManifest = {steps: [{title: ''}]};
+
   /**
    * Whether the tutorial is currently in its "solved" state.
    */
   private _solved = false;
 
+  private get _projectLocation() {
+    // TODO(e111077): delete this once we go public with tutorials catalog
+    if (window.location.pathname === '/tutorial/') {
+      return 'intro-to-lit';
+    }
+    return window.location.pathname.replace('/tutorials/', '');
+  }
+
+  /**
+   * Playground project element generated from this.project
+   */
+  private get _project(): PlaygroundProject | undefined {
+    if (!this.project) {
+      return undefined;
+    }
+    return (this.getRootNode() as ShadowRoot | Document).getElementById(
+      this.project
+    ) as PlaygroundProject | undefined;
+  }
+
+  /**
+   * Generates an info from the _idx. May result in a 404 request if the index
+   * does not result in an actual file location
+   */
+  private get _info() {
+    return this._idxToInfo(this._idx);
+  }
+
+  /**
+   * Info for the next step. May result in a 404 request if the index + 1
+   * does not result in an actual file location
+   */
+  private get _nextInfo() {
+    return this._idxToInfo(this._idx + 1);
+  }
+
+  /**
+   * Fetches the tutorial manifest on projectLocation change
+   */
+  private _manifestTask = new Task(
+    this,
+    async ([projectLocation]) => {
+      const manifestRes = await fetch(
+        `${samplesRoot}/js/tutorials/${projectLocation}/manifest.json`
+      );
+
+      const manifest = (await manifestRes.json()) as TutorialManifest;
+      this._manifest = manifest;
```

Comment:
I think there could be a race condition here?

1. _projectLocation changes to A
2. task updates and starts fetching manifest A
3. _projectLocation changes to B
4. task updates and starts fetching manifest B
5. manifest B comes back, this._manifest is set to B
6. manifest A comes back, this._manifest is set to A <-- bad!

Maybe the solution is just to check `projectLocation === this._projectLocation` before re-assigning `this._manifest`?

It sort of feels like setting state from a task body might generally be an anti-pattern, because of cases like this, wdyt? (TBH this is the first time I've looked at the task API, so I don't know what the good usage patterns are yet)